packages/promises/promise.pony

```diff
@@ -159,6 +160,88 @@ actor Promise[A: Any #share]
     _attach(consume attach)
     promise
 
+  fun tag add[B: Any #share = A](p: Promise[B]): Promise[(A, B)] =>
+    """
+    Add two promises into one promise that returns the result of both when
```

Comment:
`join` combines multiple promises of the same type and returns their final values in an array where they are in an unknown order. `add` combines promises that may yield values of different types, but even if they are the same type the promise that each value came from is known based on the order that it appears in the final tuple.